pkg/*: introduce cloudScope field to metric metadata

Abhinav1299 opened the pull request:
This commit adds cloudScope field to metric metadata to indicate where a metric is 
exported in CockroachDB Cloud (Advanced, Tenant/Serverless, or Shared). Moreover, 
tagged few metrics that are shared across offerings as `ESSENTIAL_SHARED`.

This change aims to standardize the metric visibility and enhance monitoring 
capabilities in cloud deployments.

Part of: CC-33737
Epic: CC-33357
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/153342)
<!-- Reviewable:end -->

dhartunian commented on pkg/ts/catalog/chart_catalog.proto:
```diff
@@ -96,6 +97,8 @@ message ChartMetric {
   // labeledName is the name of the metric with its labels, formatted as name{label1: value1, ...}.
   optional string labeledName = 9 [(gogoproto.nullable) = false];
+  // scope indicates the availability of this metric across different deployment offerings.
+  optional cockroach.util.metric.Metadata.Scope scope = 10 [(gogoproto.nullable) = false];
```
We should be careful to reconcile `scope` with the `essential` label because there might be some duplicated information there. All `essential: true` metrics, should automatically be `SHARED`. Anyway, maybe we can merge `essential` into this enum.

Abhinav1299 replied:
> All essential: true metrics, should automatically be SHARED.

I've crosscheck this, definition of `essential` here is specific to SH clusters. Since the goal of this task is to improve downstream visibility, `essential` cloud not be merged in `CloudScope`.

Abhinav1299 committed:
pkg/*: introduce cloudScope field to metric metadata

This commit adds cloudScope field to metric metadata to indicate where a metric
is exported in CockroachDB Cloud (Advanced, Tenant/Serverless, or Shared). Moreover,
tagged few metrics that are shared across offerings as `ESSENTIAL_SHARED`.

This change aims to standardize the metric visibility and enhance monitoring
capabilities in cloud deployments.

Part of: CC-33737
Epic: CC-33357
Release note: None

Abhinav1299 commented:
Closing this PR in light of change in approach.
We're now decoupling cloud scope with CRDB.
